src/component/ItemDetails.js

```diff
@@ -1,13 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FirestoreCollection, withFirestore } from 'react-firestore';
 import dayjs from 'dayjs';
-import { useParams } from 'react-router-dom';
+import { useParams, Redirect } from 'react-router-dom';
 import BackButton from './BackButton';
 
-const ItemDetails = ({ token, purchased }) => {
-  let { itemId } = useParams();
+const ItemDetails = ({ token, purchased, firestore }) => {
+  const [redirect, setRedirect] = useState(false);
+  const [isNewItem, setIsNewItem] = useState(true);
```

Comment:
Nice clean up with this code. Just double checking, is this state used for Delete Item? Or was it just a clean up? Want to make sure I'm not misunderstanding the actual delete part ha.

Reply:
My understanding is that `const { itemId } = useParams()` is used for the delete functionality as a way to specify what item has been selected, and the two uses of state are for sending us back to sending us back to the list afterwards and for tracking if an item has been purchased, respectively. @mindyzwan am I getting that right?

Reply:
@gunderodd - the `const { itemID } = useParams()` was actually from last week and it's where we get the reference for the specific item we want to see. The link to the item includes the ID in the url parameters, and this pulls that ID from the parameters so we can use it in our code. I must have moved it at some point 🤷‍♀ which is why it's showing up as green/red here. 

The `redirect` state is for the delete item part so we don't try to load a page for an item that doesn't exist anymore after deleting it. 

The `isNewItem` part is some cleanup we needed so we could still have the option of deleting a brand new option. Previously, we'd been getting errors when we tried to load an item detail page for an item if it had only been added to the list and not purchased yet (it didn't like the last purchase date and next purchase date part)